bin/node/src/commands.rs

```diff
@@ -0,0 +1,166 @@
+        &self,
+        keystore: &LocalKeystore,
+        key_type: KeyTypeId,
+    ) -> Result<AuraId, Error> {
+        let key = SyncCryptoStore::sr25519_public_keys(&*keystore, key_type)
+            .into_iter()
+            .next()
+            .map_or_else(
+                || {
+                    SyncCryptoStore::sr25519_generate_new(&*keystore, key_type, None)
+                        .map_err(|_| Error::KeyStoreOperation)
+                },
+                Ok,
+            );
+
+        match key {
+            Ok(k) => {
+                let bytes = k.as_array_ref();
+                Ok(AuraId::from(sr25519::Public::from_raw(*bytes)))
+            }
+            Err(why) => panic!("{}", why),
+        }
+    }
+
+    fn upsert_aleph_key(
+        &self,
+        keystore: &LocalKeystore,
+        key_type: KeyTypeId,
+    ) -> Result<AlephId, Error> {
+        let key = SyncCryptoStore::ed25519_public_keys(&*keystore, key_type)
+            .into_iter()
+            .next()
+            .map_or_else(
+                || {
+                    SyncCryptoStore::ed25519_generate_new(&*keystore, key_type, None)
+                        .map_err(|_| Error::KeyStoreOperation)
+                },
+                Ok,
+            );
+
+        match key {
+            Ok(k) => {
+                let bytes = k.as_array_ref();
+                Ok(AlephId::from(ed25519::Public::from_raw(*bytes)))
+            }
+            Err(why) => panic!("{}", why),
+        }
+    }
+
+    pub fn run(&self, cli: &Cli) -> Result<(), Error> {
+        let chain_id = self.chain_params.chain_id();
+
+        let mut genesis_authorities: Vec<AuthorityKeys> = Vec::new();
+
+        LOCAL_AUTHORITIES.iter().for_each(|authority| {
+            let authority_keystore = self
+                .open_keystore(authority, &chain_id)
+                .unwrap_or_else(|_| panic!("Cannot open keystore for {}", authority));
+
+            let aura_key = self
+                .upsert_aura_key(&authority_keystore, key_types::AURA)
+                .unwrap();
+            let aleph_key = self
+                .upsert_aleph_key(&authority_keystore, aleph_primitives::KEY_TYPE)
+                .unwrap();
+            let account_id = get_account_id_from_seed::<sr25519::Public>(authority);
+
+            // NOTE: we could generate libp2p secrets here and add PeerIds to bootstrap nodes list,
+            // see Substrate's GenerateNodeKeyCmd
+
+            genesis_authorities.push(AuthorityKeys {
+                account_id,
+                aura_key,
+                aleph_key,
+            });
+        });
+
+        info!("Building chain spec");
+
+        let chain_spec = match self.chain_params.chain_id().as_str() {
+            chain_spec::DEVNET_ID => {
+                chain_spec::development_config(self.chain_params.clone(), genesis_authorities)
+            }
+            _ => chain_spec::config(self.chain_params.clone(), genesis_authorities, &chain_id),
```

Comment:
I don't like that you run a testnet by passing as chain_id anything other than `DEVNET_ID` (and the value of the id is ignored). Can we add a `TESTNET_ID` constant , and panic when the chain_id is anything else than `DEVNET_ID` or `TESTNET_ID`?

Reply:
Sorry, but I don't follow the above two comments - we don't have a testnet config as it stands, the devnet is the only "special" kind of config created when the chain-id passed equals "dev" - this is in line with the vanilla BuildSpec command of Substrate (they have an is_dev() method on the command struct).